prov/gni/src/gnix_msg.c

```diff
@@ -1754,17 +1760,23 @@ static int __smsg_eager_msg_w_data(void *data, void *msg)
 			   req->msg.cum_send_len);
 
 		__gnix_msg_copy_data_to_recv_addr(req, data_ptr);
+
+		if ((req->msg.recv_flags & FI_MULTI_RECV) &&
+		    ((req->msg.cum_recv_len - req->msg.cum_send_len) <
+			ep->min_multi_recv))
+			req->msg.recv_flags &= ~GNIX_MSG_MULTI_RECV_SUP;
+
 		__gnix_msg_recv_completion(ep, req);
 
 		/* Check if we're using FI_MULTI_RECV and there is space left
 		 * in the receive buffer. */
 		if ((req->msg.recv_flags & FI_MULTI_RECV) &&
 		    ((req->msg.cum_recv_len - req->msg.cum_send_len) >=
-		     ep->min_multi_recv)) {
+			     ep->min_multi_recv)) {
 			GNIX_DEBUG(FI_LOG_EP_DATA, "Re-using req: %p\n", req);
 
 			/* Adjust receive buffer for the next match. */
-			req->msg.recv_info[0].recv_addr += req->msg.send_info[0].send_len;
+			req->msg.recv_info[0].recv_addr += req->msg.cum_send_len;
```

Comment:
do you mean use of ```cum_send_len```?  I had to use this because above only the ```req->msg.cum_send_len``` field was being set,  the original code with ```req->msg.send_info[0].send_len``` didn't work since it was zero.  The criterion test was failing because of the original code using ```send_info[0].send_len```.